pkg/poll/poll.go

```diff
@@ -427,14 +427,29 @@ func (p *pollActions) forBuild(ctx context.Context, obj *unstructured.Unstructur
 		return errors.Wrap(err, "Could not get BuildList")
 	}
 
-	for _, build := range builds.Items {
-		callback := makeStatusCallback("Complete", "status", "phase")
-		if err := p.forResourceFullAvailability(ctx, &build, callback); err != nil {
+	var build *unstructured.Unstructured
```

Comment:
I guess there is no object for buildconfig in the k8s client right?!

Reply:
What if we'd use https://github.com/openshift/api/blob/master/build/v1/types.go#L940 ?
`kubeClient.Get` takes a `client.Object` which `buildv1.BuildConfig{,List}` implements, also `buildv1` scheme is already installed.

Reply:
Doing this with the actual objects involves many deep changes. Will do it in another PR, as those are unrelated to the purpose of this one. Thanks!